File: hivepulse-api/internal/domain/notification.go
package domain

import "time"

type ChannelType string

const (
	ChannelEmail   ChannelType = "email"
	ChannelWebhook ChannelType = "webhook"
	ChannelSlack   ChannelType = "slack"
)

type NotificationChannel struct {
	ID                string
	Name              string
	Type              ChannelType
	Config            map[string]string
	IsGlobal          bool
	Enabled           bool
	RemindIntervalMin int
	CreatedAt         time.Time
	UpdatedAt         time.Time
}

type NotificationEvent string

const (
	EventDown      NotificationEvent = "down"
	EventUp        NotificationEvent = "up"
	EventSSLExpiry NotificationEvent = "ssl_expiry"
)

type NotificationLog struct {
	ID        int64
	ChannelID string
	MonitorID string
	Event     NotificationEvent
	Status    string // "sent" | "failed"
	ErrorMsg  string
	SentAt    time.Time
}

// ReminderTarget is a (monitor, channel) pair eligible for re-notification.
type ReminderTarget struct {
	MonitorID string
	ChannelID string
}

File: hivepulse-web/src/domain/notification.ts
export type ChannelType = 'email' | 'webhook' | 'slack'
export type NotificationEvent = 'down' | 'up' | 'ssl_expiry'

export interface NotificationChannel {
  id: string
  name: string
  type: ChannelType
  config: Record<string, string>
  is_global: boolean
  enabled: boolean
  remind_interval_min: number
  created_at: string
}

export interface NotificationLog {
  id: number
  channel_id: string
  monitor_id: string
  event: NotificationEvent
  status: 'sent' | 'failed'
  error_msg: string
  sent_at: string
}

export interface CreateChannelInput {
  name: string
  type: ChannelType
  config: Record<string, string>
  is_global: boolean
  enabled: boolean
  remind_interval_min: number
}

export interface ScheduleRule {
  days: string[]  // "mon" | "tue" | "wed" | "thu" | "fri" | "sat" | "sun"
  start: string   // "HH:MM"
  end: string     // "HH:MM"
}

export interface AssignmentTriggers {
  cooldown_minutes: number
  schedule?: ScheduleRule
}

export interface MonitorChannelAssignment {
  id: string
  name: string
  type: ChannelType
  triggers: AssignmentTriggers
}
